### PR TITLE
ci: add dependency on wait-30s job for core commit resolution

### PR DESCRIPTION
**Pull request by jamil314**
## Description

Clearly describe what has been changed. Include relevant context or background.
Explain how the issue was fixed (if applicable) and the root cause.

Link this pull request to the GitHub issue (and optionally name the branch `ocrvs-<issue #>`)

## Checklist

- [ ] I have linked the correct Github issue under "Development"
- [ ] I have tested the changes locally, and written appropriate tests
- [ ] I have tested beyond the happy path (e.g. edge cases, failure paths)
- [ ] I have updated the changelog with this change (if applicable)
- [ ] I have updated the GitHub issue status accordingly

### CODE REVIEW

File: .github/workflows/deploy-and-e2e.yml [hunk 1]
```diff
@@ -32,13 +32,14 @@ jobs:
...
-    name: wait 30s
+    name: Wait 30 seconds
     runs-on: ubuntu-latest
     steps:
-      - name: wait 30s
+      - name: Wait 30 seconds
         run: sleep 30
 
   get-core-commit:
+    needs: wait-30s
```
Comment (Zangetsu101) [inline .github/workflows/deploy-and-e2e.yml:42]: Shouldn't we wait on the "resolve cc commit" too?

Comment (jamil314) [reply .github/workflows/deploy-and-e2e.yml:42]: It already waits for core's resolve

Comment (jamil314) [reply .github/workflows/deploy-and-e2e.yml:42]:
```
  get-country-config-commit:
    name: Resolve latest Farajaland tag
    runs-on: ubuntu-latest
    needs: get-core-commit
```

**Review by makelicious — APPROVED** [review]

**Comment by jamil314** [comment]
[e2e passed](https://github.com/opencrvs/e2e/actions/runs/27533622797)
